### PR TITLE
Update DG UML diagrams. Edit DG.

### PR DESCRIPTION
**Pull request by Effixion**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,10 +1,6 @@
----
-layout: page
-title: Developer Guide
----
+# **Developer Guide**
```
Comment (avock) [inline docs/DeveloperGuide.md:1]: This part has to be kept for the webpage rendering

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1,10 +1,6 @@
...
 
 * Table of Contents
-  {:toc}
```
Comment (avock) [inline docs/DeveloperGuide.md:7]: Will need to add this in the future DG version, just it ignore for now

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -163,12 +159,6 @@ The `Model` component,
 * does not depend on any of the other three components (as the `Model` represents data entities of the domain, they
   should make sense on their own without depending on other components)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
-
-<img src="images/BetterModelClassDiagram.png" width="450" />
-
-</div>
-
```
Comment (avock) [inline docs/DeveloperGuide.md:171]: Is this removed as we're not including suggestions/ideas?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -319,127 +302,36 @@ Given below is an example usage scenario, and an object diagram to show the obje
...
+![AddModuleObjectDiagram](images/AddModuleObjectDiagram.png)
 
 The following sequence diagram shows how the AddModule operation works:
-(to be added)
-[//]: # (![AddModuleSequenceDiagram]&#40;images/AddModuleSequenceDiagram.png&#41;)
-
-
-### \[Proposed\] Undo/redo feature
-
-#### Proposed Implementation
```
Comment (avock) [inline docs/DeveloperGuide.md:345]: I think this part can be kept here, we can just add features that are more relevant to SoCompiler

**Review by avock — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Effixion) [reply docs/DeveloperGuide.md:171]: I think I don't really want to include this since I didn't really want to update the UML diagram.

**Review by Yingming23 — APPROVED** [review]

**Review by avock — APPROVED** [review]

**Pull request merged**